modules/samples/authenticators/components/org.wso2.carbon.identity.sample.extension.auth.endpoint/src/main/webapp/fpt.jsp

```diff
@@ -0,0 +1,149 @@
+<%--
+  ~ Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+  ~
+  ~ WSO2 Inc. licenses this file to you under the Apache License,
+  ~ Version 2.0 (the "License"); you may not use this file except
+  ~ in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  --%>
+<%@ page language="java" session="true" %>
+<%@ page import="java.net.URLDecoder" %>
+<%@ page import="java.nio.charset.StandardCharsets" %>
+<%@ page import="org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils" %>
+<%
+
+    String fingerprint = request.getParameter("fingerprint");
+    String callbackUrl = request.getParameter("callbackUrl");
+
+    if (fingerprint != null) {
+        if (callbackUrl != null) {
```

Comment:
Is there any condition why you have split these two conditions to two statements ? You can evaluate these two at the same statement